LinearRegression/linear_regression_blog.md

```diff
@@ -0,0 +1,718 @@
+std_error_elastic_net = scores_std_elastic_net / np.sqrt(X_.shape[0])
+# shade the region within 1 std err
+plt.semilogx(alphas_elastic_net, scores_elastic_net + std_error_elastic_net, 'b--')
+plt.semilogx(alphas_elastic_net, scores_elastic_net - std_error_elastic_net, 'b--')
+plt.fill_between(alphas_elastic_net, scores_elastic_net + std_error_elastic_net, scores_elastic_net - std_error_elastic_net, alpha = 0.25)
+plt.ylabel('CV score +/- std error')
+plt.xlabel('Log lambda')
+plt.axhline(np.min(scores_elastic_net), linestyle='--', color='.5')
+plt.xlim([alphas_elastic_net[0], alphas_elastic_net[-1]])
+plt.title('CV score as a function of lambda')
+plt.show()
+# print the optimal values
+print('best score:', np.negative(cv_elastic_net.best_score_))
+print('lambda:', cv_elastic_net.best_params_.get('alpha'))
+
+# print the standardized coefficients
+print('(Standardized)')
+elastic = linear_model.ElasticNet(alpha = cv_elastic_net.best_params_.get('alpha'))
+elastic.fit(X_std, y)
+elastic_df = pd.DataFrame(elastic.coef_, X_.columns, columns=['Coefficient'])
+print('Intercept                                 %.6f' %(elastic.intercept_))
+print(elastic_df)
+
+# back-convert for ease of interpretation
+# beta_real = beta_std / std err
+# intercept_real = intercept_std - sum of (mean * beta_std / stderr)
+print('\n(Real)')
+elastic_real = elastic_df.copy()
+elastic_intercept_real = elastic.intercept_
+count_elastic = elastic_real.shape[0]
+for i in range(count_elastic):
+  elastic_real.iloc[i, 0] /= np.std(X_.iloc[:, i])
+  elastic_intercept_real -= (elastic_real.iloc[i, 0] * np.mean(X_.iloc[:, i]))
+print('Intercept                             %.6f' %(elastic_intercept_real))
+print(elastic_real)
+```
+
+<img src="Blog/47.png" style="float: left;" />
+
+### Comparing Performances
+
+Again, we can calculate the adjusted $R^2$ of each of these regressions. You can see that their values are all around 0.576, which is better than the adjusted $R^2$ of our initial OLS model.
+
+```python
+y_pred_lasso = lasso.predict(X_std) # predictions
+r2_lasso = metrics.r2_score(y, y_pred_lasso) # R-squared
+adj_r2_lasso = 1 - (1 - r2_lasso) * (X_.shape[0] - 1) / (X_.shape[0] - count_lasso - 1) # adjusted R-squared
+print('Lasso adj. R2:', adj_r2_lasso)
+
+y_pred_ridge = ridge.predict(X_std) # predictions
+r2_ridge = metrics.r2_score(y, y_pred_ridge) # R-squared
+adj_r2_ridge = 1 - (1 - r2_ridge) * (X_.shape[0] - 1) / (X_.shape[0] - count_ridge - 1) # adjusted R-squared
+print('Ridge adj. R2:', adj_r2_ridge)
+
+y_pred_elastic = elastic.predict(X_std) # predictions
+r2_elastic = metrics.r2_score(y, y_pred_elastic) # R-squared
+adj_r2_elastic = 1 - (1 - r2_elastic) * (X_.shape[0] - 1) / (X_.shape[0] - count_elastic - 1) # adjusted R-squared
+print('Elastic Net adj. R2:', adj_r2_elastic)
+```
+
+<img src="Blog/49.png" style="float: left;" />
+
+
+
+That’s it for a complete procedure to perform linear regression analysis in Python! We have
```

Comment:
This is a solid draft- I think making some of the longer paragraphs shorter would be great. Very organized and put together!